demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -1183,6 +1191,24 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
 
     if( ( xResult == CKR_OK ) && ( CK_TRUE == xKeyPairGenerationMode ) )
     {
+        /* Add a delay before calling logic for generating new key-pair on device (if boards supports on-board key-pair
+         * generation) to avoid possible scenario of unexpectedly generating new keys on board during the flashing process
+         * of a new image on the board.
+         * If the flashing workflow of a device (for example, ESP32 boards) involves resetting the board before
+         * flashing a new image, then a race condition can occur between the execution of an already
+         * existing image on device (that is triggered by the device reset) and the flashing of the new image on the
+         * device. When the existing image present on the device is configured to generate new key-pair (through the
+         * keyprovisioningFORCE_GENERATE_NEW_KEY_PAIR config), then a possible scenario of unexpected key-pair
+         * generation on device can occur during flashing process, in which case, the certificate provisioned by
+         * user becomes stale and device cannot perform TLS connection with servers as the provisioned device certificate
+         * does not match the unexpectedly generated new key-pair.
+         * Thus, by adding a delay, the possibility of hitting the race condition of the device executing an old
+         * image that generates new key-pair is avoided because the logic of generating new key-pair is not executed
+         * before the flashing process starts loading the new image onto the board.
+         * Note: The delay of 5 seconds is used based on testing with an ESP32+ECC608A board. */
```

Comment:
The comment mentioned 5 secs used testing with ESP32+ECC608A. The default sets to 30 secs. Is there any particular reason for the magic 30?

Reply:
Fixed to mention 30 seconds